### PR TITLE
feat: council arbitrator

### PR DESCRIPTION
**Pull request by 0xJabberwock**
# 🤖 Linear

Closes GRT-47

### CODE REVIEW

File: src/contracts/CouncilArbitrator.sol [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ARBITRATOR_MODULE = _arbitratorModule; // TODO: IModule should be IValidator
+  }
+
+  /// @inheritdoc IArbitrator
+  function resolve(
+    IOracle.Request memory _request,
+    IOracle.Response memory _response,
+    IOracle.Dispute memory _dispute
+  ) external onlyArbitratorModule returns (bytes memory /* _data */ ) {
+    // TODO: Emit event with params? Save params locally?
```
Comment (0xShaito) [inline src/contracts/CouncilArbitrator.sol:56]: I would emit event and save params locally to validate the call in resolveDispute and see that the disputeId is correct

File: src/contracts/CouncilArbitrator.sol [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  /// @inheritdoc ICouncilArbitrator
+  function resolveDispute(bytes32 _disputeId, IOracle.DisputeStatus _status) external onlyArbitrator {
+    // TODO: Checks?
+
+    getAnswer[_disputeId] = _status;
+
+    // TODO: Provide or fetch params?
```
Comment (0xShaito) [inline src/contracts/CouncilArbitrator.sol:65]: We can save them from the call to resolve

File: src/contracts/CouncilArbitrator.sol (hunk 1 above)
Comment (0xShaito) [inline src/contracts/CouncilArbitrator.sol:56]: We need to also check that the _dispute + _response + _dispute is a valid combo

**Review by 0xShaito — CHANGES_REQUESTED** [review]

File: src/contracts/CouncilArbitrator.sol (hunk 1 above)
Comment (0xShaito) [reply src/contracts/CouncilArbitrator.sol:56]: disputeId -> _request, _response, _dispute

Comment (0xShaito) [inline src/contracts/CouncilArbitrator.sol:56]: Emit event + save params locally disputeId -> _request, _response, _dispute

**Review by 0xShaito — APPROVED** [review]

**Review by ashitakah — APPROVED** [review]

**Pull request merged**